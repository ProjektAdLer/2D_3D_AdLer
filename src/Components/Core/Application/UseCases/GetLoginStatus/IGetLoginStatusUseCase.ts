import { ISynchronousUsecase } from "../../Abstract/ISynchronousUsecase";
import LoginStatusTO from "../../DataTransferObjects/LoginStatusTO";

export default interface IGetLoginStatusUseCase
  extends ISynchronousUsecase<void, LoginStatusTO> {}
