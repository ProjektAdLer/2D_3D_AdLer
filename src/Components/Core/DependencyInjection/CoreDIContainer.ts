import { LoadWorldController } from "../Presentation/React/LoadWorldButton/LoadWorldController";
import ILoadWorldController from "../Presentation/React/LoadWorldButton/ILoadWorldController";
import { Container } from "inversify";
import CORE_TYPES from "./CoreTypes";
import ISceneView from "../Presentation/Babylon/SceneManagement/ISceneView";
import SceneView from "../Presentation/Babylon/SceneManagement/SceneView";
import SceneViewModel from "../Presentation/Babylon/SceneManagement/SceneViewModel";
import ScenePresenter from "../Presentation/Babylon/SceneManagement/ScenePresenter";
import MainScene from "../Presentation/Babylon/SceneManagement/MainScene";
import ICreateSceneClass from "../Presentation/Babylon/SceneManagement/ICreateSceneClass";
import IScenePresenter from "../Presentation/Babylon/SceneManagement/IScenePresenter";
import infrastructureDIContainer from "./InfrastructureDIContainer";
import useCaseDIContainer from "./UseCases/UseCaseDIConatiner";
import PortsDIContainer from "./Ports/PortsDIContainer";
import BuilderDIContainer from "./Builders/BuilderDIContainer";
import INavigation from "../Presentation/Babylon/Navigation/INavigation";
import Navigation from "../Presentation/Babylon/Navigation/Navigation";
import NavigationConfiguration from "../Presentation/Babylon/Navigation/NavigationConfiguration";

var CoreDIContainer = new Container();

// Scene
CoreDIContainer.bind<ISceneView>(CORE_TYPES.ISceneView)
  .to(SceneView)
  .inSingletonScope();
CoreDIContainer.bind(SceneViewModel).toSelf().inSingletonScope();
CoreDIContainer.bind<IScenePresenter>(CORE_TYPES.IScenePresenter)
  .to(ScenePresenter)
  .inSingletonScope();

// Navigation
CoreDIContainer.bind<INavigation>(CORE_TYPES.INavigation)
  .to(Navigation)
  .inSingletonScope();
CoreDIContainer.bind(NavigationConfiguration).toSelf();

// Controllers
CoreDIContainer.bind<ILoadWorldController>(CORE_TYPES.ILoadWorldController)
  .to(LoadWorldController)
  .inSingletonScope();

// bind other CreateSceneClass here for testing puposes -MK
CoreDIContainer.bind<ICreateSceneClass>(CORE_TYPES.ICreateSceneClass).to(
  MainScene
);

CoreDIContainer.load(infrastructureDIContainer);
CoreDIContainer.load(useCaseDIContainer);
CoreDIContainer.load(PortsDIContainer);
CoreDIContainer.load(BuilderDIContainer);

export default CoreDIContainer;
