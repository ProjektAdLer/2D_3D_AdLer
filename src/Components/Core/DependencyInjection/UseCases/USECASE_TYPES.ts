const USECASE_TYPES = {
  ILoadUserLearningWorldsUseCase: Symbol("ILoadUserLearningWorldsUseCase"),
  ILoadLearningWorldUseCase: Symbol("ILoadLearningWorldUseCase"),
  ILoadAvatarUseCase: Symbol("ILoadAvatarUseCase"),
  ILoadLearningElementUseCase: Symbol("ILoadLearningElementUseCase"),
  IScoreLearningElementUseCase: Symbol("IScoreLearningElementUseCase"),
  ICalculateLearningSpaceScoreUseCase: Symbol(
    "ICalculateLearningSpaceScoreUseCase"
  ),
  ICalculateLearningWorldScoreUseCase: Symbol(
    "ICalculateLearningWorldScoreUseCase"
  ),
  ILoginUseCase: Symbol("ILoginUseCase"),
  ILoadLearningSpaceUseCase: Symbol("ILoadLearningSpaceUseCase"),
  IScoreH5PLearningElementUseCase: Symbol("IScoreH5PLearningElementUseCase"),
  IGetLearningElementSourceUseCase: Symbol("IGetLearningElementSourceUseCase"),
  IGetLoginStatusUseCase: Symbol("IGetLoginStatusUseCase"),
  ISetUserLocationUseCase: Symbol("ISetUserLocationUseCase"),
  IGetUserLocationUseCase: Symbol("IGetUserLocationUseCase"),
  ICalculateLearningSpaceAvailabilityUseCase: Symbol(
    "ICalculateLearningSpaceAvailabilityUseCase"
  ),
  IGetLearningSpacePrecursorAndSuccessorUseCase: Symbol(
    "IGetLearningSpacePrecursorAndSuccessorUseCase"
  ),
  ILoadAdaptivityElementUseCase: Symbol("ILoadAdaptivityElementUseCase"),
  IGetAdaptivityElementStatusUseCase: Symbol(
    "IGetAdaptivityElementStatusUseCase"
  ),
  ISubmitAdaptivityElementSelectionUseCase: Symbol(
    "ISubmitAdaptivityElementSelectionUseCase"
  ),
  IStartOverallTimeSpentNotificationTimerUseCase: Symbol(
    "IStartOverallTimeSpentNotificationTimerUseCase"
  ),
  ICreateOverallTimeSpentNotificationTimerUseCase: Symbol(
    "ICreateOverallTimeSpentNotificationTimerUseCase"
  ),
  IPauseOverallTimeSpentNotificationTimerUseCase: Symbol(
    "IPauseOverallTimeSpentNotificationTimerUseCase"
  ),
};

export default USECASE_TYPES;
