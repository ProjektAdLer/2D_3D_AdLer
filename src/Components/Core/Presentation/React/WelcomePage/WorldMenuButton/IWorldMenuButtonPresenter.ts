export default interface IWorldMenuButtonPresenter {
  setLoginSuccessful(): void;
}
