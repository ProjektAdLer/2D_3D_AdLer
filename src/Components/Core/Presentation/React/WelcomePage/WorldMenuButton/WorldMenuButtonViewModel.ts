import Observable from "../../../../../../Lib/Observable";

export default class WorldMenuButtonViewModel {
  userLoggedIn = new Observable<boolean>(false, true);
}
