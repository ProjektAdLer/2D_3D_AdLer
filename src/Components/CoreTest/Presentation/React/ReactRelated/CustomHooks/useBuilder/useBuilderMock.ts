import * as useBuilder from "~ReactComponents/ReactRelated/CustomHooks/useBuilder";

/**
 * Mocks the useViewModelControllerProvider hook.
 * @param mockedTupels The return value, which the mocked hook will return.
 *
 */
export default function useBuilderMock(
  mockedTupels: [viewModel: any, controller: any] = [undefined, undefined]
) {
  jest.spyOn(useBuilder, "default").mockReturnValue(mockedTupels);
}
