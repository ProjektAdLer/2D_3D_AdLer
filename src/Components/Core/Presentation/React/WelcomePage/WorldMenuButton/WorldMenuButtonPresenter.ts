import IWorldMenuButtonPresenter from "./IWorldMenuButtonPresenter";
import WorldMenuButtonViewModel from "./WorldMenuButtonViewModel";

export default class WorldMenuButtonPresenter
  implements IWorldMenuButtonPresenter
{
  constructor(private viewModel: WorldMenuButtonViewModel) {}
  onLoginSuccessful(): void {
    this.viewModel.loggedInMoodle.Value = true;
  }
}
