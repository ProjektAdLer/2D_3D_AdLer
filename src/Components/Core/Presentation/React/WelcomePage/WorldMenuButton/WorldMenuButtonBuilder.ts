import { injectable } from "inversify";
import PresentationBuilder from "../../../PresentationBuilder/PresentationBuilder";
import CoreDIContainer from "../../../../DependencyInjection/CoreDIContainer";
import ILMSPort from "../../../../Ports/LMSPort/ILMSPort";
import PORT_TYPES from "../../../../DependencyInjection/Ports/PORT_TYPES";
import WorldMenuButtonViewModel from "./WorldMenuButtonViewModel";
import WorldMenuButtonPresenter from "./WorldMenuButtonPresenter";

@injectable()
export default class WorldMenuButtonBuilder extends PresentationBuilder<
  WorldMenuButtonViewModel,
  undefined,
  undefined,
  WorldMenuButtonPresenter
> {
  constructor() {
    super(
      WorldMenuButtonViewModel,
      undefined,
      undefined,
      WorldMenuButtonPresenter
    );
  }

  override buildPresenter(): void {
    super.buildPresenter();
    CoreDIContainer.get<ILMSPort>(
      PORT_TYPES.IMoodlePort
    ).registerWorldMenuButtonPresenter(this.presenter!);
  }
}
