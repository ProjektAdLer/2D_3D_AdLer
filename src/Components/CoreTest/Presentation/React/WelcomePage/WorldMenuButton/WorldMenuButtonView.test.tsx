import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import React from "react";
import useBuilderMock from "../../ReactRelated/CustomHooks/useBuilder/useBuilderMock";
import WorldMenuButton from "../../../../../Core/Presentation/React/WelcomePage/WorldMenuButton/WorldMenuButtonView";
import WorldMenuButtonViewModel from "../../../../../Core/Presentation/React/WelcomePage/WorldMenuButton/WorldMenuButtonViewModel";
import history from "history/browser";
import { Provider } from "inversify-react";
import CoreDIContainer from "../../../../../Core/DependencyInjection/CoreDIContainer";
import { mock } from "jest-mock-extended";
import IGetLoginStatusUseCase from "../../../../../Core/Application/UseCases/GetLoginStatus/IGetLoginStatusUseCase";
import USECASE_TYPES from "../../../../../Core/DependencyInjection/UseCases/USECASE_TYPES";

let mockViewModel = new WorldMenuButtonViewModel();
const historyPushMock = jest.spyOn(history, "push");
const getLoginStatusUseCaseMock = mock<IGetLoginStatusUseCase>();

describe("WorldMenuButton", () => {
  beforeAll(() => {
    CoreDIContainer.snapshot();
    CoreDIContainer.rebind<IGetLoginStatusUseCase>(
      USECASE_TYPES.IGetLoginStatusUseCase
    ).toConstantValue(getLoginStatusUseCaseMock);
  });

  afterAll(() => {
    CoreDIContainer.restore();
  });

  test("should render", () => {
    useBuilderMock([mockViewModel, undefined]);
    render(
      <Provider container={CoreDIContainer}>
        <WorldMenuButton />
      </Provider>
    );
  });

  test("LoginComponent calls GetLoginStatusUseCase.execute on mount", () => {
    useBuilderMock([mockViewModel, undefined]);

    render(
      <Provider container={CoreDIContainer}>
        <WorldMenuButton />
      </Provider>
    );

    expect(getLoginStatusUseCaseMock.execute).toBeCalledTimes(1);
  });

  test("WorldMenuButton Tailwind Styling contains grey backgroundColor if not logged in", () => {
    mockViewModel.userLoggedIn.Value = false;
    getLoginStatusUseCaseMock.execute.mockReturnValue(false);
    useBuilderMock([mockViewModel, undefined]);

    const componentUnderTest = render(
      <Provider container={CoreDIContainer}>
        <WorldMenuButton />
      </Provider>
    );

    const style =
      componentUnderTest.container.children[0].children[0].className;
    expect(style).toContain("bg-adlerdeactivated");
  });

  test("WorldMenuButton Tailwind Styling contains normal backgroundColor if logged in", () => {
    mockViewModel.userLoggedIn.Value = true;
    getLoginStatusUseCaseMock.execute.mockReturnValue(true);
    useBuilderMock([mockViewModel, undefined]);

    const componentUnderTest = render(
      <Provider container={CoreDIContainer}>
        <WorldMenuButton />
      </Provider>
    );

    const style =
      componentUnderTest.container.children[0].children[0].className;
    expect(style).toContain("bg-adlerblue");
  });

  test("on click calls history.push", () => {
    mockViewModel.userLoggedIn.Value = true;
    getLoginStatusUseCaseMock.execute.mockReturnValue(true);
    useBuilderMock([mockViewModel, undefined]);

    const componentUnderTest = render(
      <Provider container={CoreDIContainer}>
        <WorldMenuButton />
      </Provider>
    );

    let button: HTMLElement = componentUnderTest.getByRole("button");

    fireEvent.click(button);

    expect(historyPushMock).toHaveBeenCalledTimes(1);
  });
});
