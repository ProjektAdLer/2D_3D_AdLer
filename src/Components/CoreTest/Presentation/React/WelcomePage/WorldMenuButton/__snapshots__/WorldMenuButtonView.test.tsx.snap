// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorldMenuButton WorldMenuButton Tailwind Styling contains grey backgroundColor if not logged in 1`] = `"text-adlerdeactivatedtext bg-adlerbuttonlocked hover:cursor-default rounded-lg p-2"`;

exports[`WorldMenuButton WorldMenuButton Tailwind Styling contains normal backgroundColor if logged in 1`] = `"px-2 py-1 flex items-center text-sm rounded-lg hover:cursor-pointer lg:text-xl active:translate-x-1 active:translate-y-1 active:border-transparent text-adlerdarkblue font-regular border-t-[1px] border-l-[1px] border-b-4 border-r-4 border-adlerdarkblue bg-buttonbgblue"`;
